chia/data_layer/download_data.py

```diff
@@ -0,0 +1,184 @@
+            serialized_node = SerializedNode.from_bytes(serialize_nodes_bytes)
+
+            node_type = NodeType.TERMINAL if serialized_node.is_terminal else NodeType.INTERNAL
+            await data_store.insert_node(node_type, serialized_node.value1, serialized_node.value2)
+
+    await data_store.insert_root_with_ancestor_table(tree_id=tree_id, node_hash=root_hash, status=Status.COMMITTED)
+
+
+async def write_files_for_root(
+    data_store: DataStore,
+    tree_id: bytes32,
+    root: Root,
+    foldername: Path,
+    override: bool = False,
+) -> bool:
+    if root.node_hash is not None:
+        node_hash = root.node_hash
+    else:
+        node_hash = bytes32([0] * 32)  # todo change
+
+    filename_full_tree = foldername.joinpath(get_full_tree_filename(tree_id, node_hash, root.generation))
+    filename_diff_tree = foldername.joinpath(get_delta_filename(tree_id, node_hash, root.generation))
+
+    written = False
+    mode: Literal["wb", "xb"] = "wb" if override else "xb"
+
+    try:
+        with open(filename_full_tree, mode) as writer:
+            await data_store.write_tree_to_file(root, node_hash, tree_id, False, writer)
+        written = True
+    except FileExistsError:
+        pass
+
+    try:
+        last_seen_generation = await data_store.get_last_tree_root_by_hash(
+            tree_id, root.node_hash, max_generation=root.generation
+        )
+        if last_seen_generation is None:
+            with open(filename_diff_tree, mode) as writer:
+                await data_store.write_tree_to_file(root, node_hash, tree_id, True, writer)
+        else:
+            open(filename_diff_tree, mode).close()
+        written = True
+    except FileExistsError:
+        pass
+
+    return written
+
+
+async def insert_from_delta_file(
+    data_store: DataStore,
+    tree_id: bytes32,
+    existing_generation: int,
+    root_hashes: List[bytes32],
+    server_info: ServerInfo,
+    client_foldername: Path,
+    log: logging.Logger,
+) -> bool:
+    for root_hash in root_hashes:
+        timestamp = int(time.time())
+        existing_generation += 1
+        filename = get_delta_filename(tree_id, root_hash, existing_generation)
+
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(server_info.url + "/" + filename) as resp:
+                    resp.raise_for_status()
+
+                    target_filename = client_foldername.joinpath(filename)
+                    text = await resp.read()
+                    target_filename.write_bytes(text)
+        except Exception:
+            await data_store.server_misses_file(tree_id, server_info, timestamp)
+            raise
+
+        log.info(f"Successfully downloaded delta file {filename}.")
+        try:
+            await insert_into_data_store_from_file(
+                data_store,
+                tree_id,
+                None if root_hash == bytes32([0] * 32) else root_hash,
```

Comment:
```suggestion
                None if root_hash == DEFAULT_ROOT_HASH else root_hash,
```

Should be a constant so you can be sure you catch them all when you replace.